src/mainwindow.cpp

```diff
@@ -219,17 +219,33 @@ void MainWindow::receiveChildParentIdxs(
   if (!animatePlanning_) {
     // render visited grids
     for (auto idx : visitedVerticesIdxOrder_) {
-      if (idx != goalKeyIdx_)
+      if (idx != goalKeyIdx_) {
         gridMap->getGridsList()->at(idx)->setBrush(
-            QBrush(Qt::blue, Qt::SolidPattern));
-    }
+            QBrush(QColor(200, 200, 200, 245), Qt::SolidPattern));
```

Comment:
Same goes to here too. How about we create some setter functions in Grid for creating visuals and call these inside this loop?

Reply:
And I think thread issue happens since this function runs in different therad, you can fix this by removing `Qt::DirectConnection` in line 343.